gitbook-docs/data_model.md

```diff
@@ -171,9 +171,10 @@ A delta message can be recognised from the other types by the topmost level havi
 `updates` is the only required property.
 If `context` is missing it is assumed that the data is related to the `self` context.
 
-Context is a path from the root of the full tree. In this case 'vessels.urn:mrn:imo:mmsi:234567890'. All subsequent data is relative to
-that location. The context could be much more specific, e.g. 'vessels.urn:mrn:imo:mmsi:234567890.navigation', whatever is the common root
-of the updated data.
+Context is a path from the root of the full tree to the _container object_, which is always a vessel with the current Signal K version.
+The delimiter in the context path is `.` (period).
```

Comment:
In the REST API the escaping mechanism is defined by the URI specification so that is already covered. We do not have an equivalent for our `.` delimited paths.

This change now says context is always a vessel whereas before it allowed any node in the model. That is not a compatible change.

It is introducing _container object_ and _leaf path_ as special concepts in the model. This is good in that it is reflecting reality better than the old hierarchy but is a fundamental change. Trying to force backward compatibility with the existing implementation is causing oddities like this one where applications have to parse `context` as a string to extract a context/id pair where there are simpler and more representative ways to convey that.